src/Components/Components/src/Routing/LocationChangingContext.cs

```diff
@@ -0,0 +1,43 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+namespace Microsoft.AspNetCore.Components.Routing;
+
+/// <summary>
+/// Contains context for a change to the browser's current location.
+/// </summary>
+public class LocationChangingContext
+{
+    private readonly CancellationTokenSource _cts;
+
+    internal LocationChangingContext(string destinationUrl, bool isNavigationIntercepted, CancellationTokenSource cts)
```

Comment:
Does this also need to receive the state?

Reply:
Yeah, that seems reasonable. I wonder if we should add this in a follow-up just to keep the initial API/implementation as simple as possible?

Reply:
I think you need to handle both at the same time, adding a feature can't break another feature.

Reply:
How would this be breaking another feature? The state will still be provided in `LocationChangedEventArgs`. I interpreted your suggestion as meaning we would also add a `public string? HistoryEntryState { get; }` property to `LocationChangingContext` so that you could see the state of the location being navigated to. Couldn't this be something we add in a follow-up without breaking other things? Let me know if I'm misinterpreting all of this 😄